### PR TITLE
Remove signature from the server response.

### PR DESCRIPTION
**Pull request by FabienChaynes**
The server extension uses the signature to validate the incoming message. But once this is done, the signature is not removed from the payload and keeps being transmitted until the response is sent, even if it's useless.

This PR removes the signature once it has been used by the server extension in order to reduce the payload size.

### CODE REVIEW

File: lib/faye/authentication/server_extension.rb [hunk 1]
```diff
@@ -26,6 +26,7 @@ def incoming(message, callback)
               end
             debug("Authentication failed: #{message['error']}")
           end
+          message.delete('signature')
```
Comment (DrissTM) [inline lib/faye/authentication/server_extension.rb]: is the signature present even if `Faye::Authentication.authentication_required?` is false ?

Comment (DrissTM) [reply lib/faye/authentication/server_extension.rb]: I think I found the answer, the JS has the same `authentication_required` check and only adds the signature when needed, am I right ?

Comment (FabienChaynes) [reply lib/faye/authentication/server_extension.rb]: > I think I found the answer, the JS has the same authentication_required check and only adds the signature when needed, am I right ?

Yup.

Basically, I don't want to mess with the payload if the extension is not supposed to be active.

Comment (DrissTM) [reply lib/faye/authentication/server_extension.rb]: Okay make sense :+1:

Comment (jarthod) [reply lib/faye/authentication/server_extension.rb]: if it's not there will it raise?

Comment (FabienChaynes) [reply lib/faye/authentication/server_extension.rb]: No.

Comment (jarthod) [reply lib/faye/authentication/server_extension.rb]: Thanks. Will be a good idea to verify after this if we still have signtures in outbound messages :)

Comment (FabienChaynes) [reply lib/faye/authentication/server_extension.rb]: I checked it locally:
Before:
Selection_438

After:
Selection_437

Comment (jarthod) [reply lib/faye/authentication/server_extension.rb]: Yes I mean if we find message with signature for which `Faye::Authentication.authentication_required?` is false

**Pull request merged**